### PR TITLE
Web app block sync UI v2

### PR DESCRIPTION
**Pull request by grod220**
Closes https://github.com/penumbra-zone/web/issues/305
Part 2 of: https://github.com/penumbra-zone/web/pull/600

Adding a more prominent block sync status to web app. 

- `LastBlockSynced` renamed to `FullSyncHeight` to align with protobuf naming
- Bumped the indexdb version so locally stored sync height reset (given new key)

### CODE REVIEW

File: apps/extension/.env [hunk 1]
```diff
@@ -1,6 +1,6 @@
 
 PRAX=lkpmkhpnhknhmibgnmmhdhgdilepfghe
-IDB_VERSION=23
+IDB_VERSION=24
```
Comment (grod220) [inline apps/extension/.env:3]: New key for sync height, so easiest to wipe state

File: apps/webapp/src/components/header/top-row.tsx
Comment (grod220) [inline apps/webapp/src/components/header/top-row.tsx:1]: Largely all moved code

File: packages/ui/lib/toast/transaction-toast.tsx [hunk 2]
```diff
@@ -49,6 +50,14 @@ export class TransactionToast {
...
   constructor(private transactionClassification: TransactionClassification) {}
 
+  private get label(): string {
+    return TRANSACTION_LABEL_BY_CLASSIFICATION[this.transactionClassification];
+  }
+
+  private get shortenedTxHash(): string {
+    return this._txHash ? shorten(this._txHash, 8) : '';
+  }
+
```
Comment (jessepinho) [inline packages/ui/lib/toast/transaction-toast.tsx:60]: Was this an automatic change from your IDE? I feel like it's generally the norm to put private methods last.

File: packages/types/src/indexed-db.ts [hunk 3]
```diff
@@ -48,7 +48,7 @@ export interface IndexedDbInterface {
   ): AsyncGenerator<IdbUpdate<DBTypes, StoreName>, void>;
   constants(): IdbConstants;
   clear(): Promise<void>;
-  getLastBlockSynced(): Promise<bigint | undefined>;
+  getfullSyncHeight(): Promise<bigint | undefined>;
```
[the remark this answers is not in the document]
Comment (jessepinho) [reply packages/types/src/indexed-db.ts:51]: (And this should be changed everywhere this method is referenced 😬 `packages/storage/src/indexed-db/index.ts`, `packages/router/src/grpc/view-protocol-server/test-utils.ts`, etc.)

File: packages/storage/src/indexed-db/index.ts [hunk 4]
```diff
@@ -139,7 +139,7 @@ export class IndexedDb implements IndexedDbInterface {
     await this.u.updateAll(txs);
   }
 
-  async getLastBlockSynced() {
+  async getfullSyncHeight() {
     return this.db.get('LAST_BLOCK_SYNCED', 'last_block');
```
Comment (jessepinho) [inline packages/storage/src/indexed-db/index.ts:143]: Shouldn't this key also be changed to `FULL_SYNC_HEIGHT`?

File: apps/webapp/src/components/header/sync-status-section.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { useStream } from '../../fetchers/stream';
+import { useMemo } from 'react';
+import { viewClient } from '../../clients/grpc';
+import { BlockSyncStatus } from '@penumbra-zone/ui';
+
+export const SyncStatusSection = () => {
```
Comment (jessepinho) [inline apps/webapp/src/components/header/sync-status-section.tsx:6]: Totally not necessary to change the UI for this PR, but my feeling is that the block sync takes up a lot of above-the-fold space for something that probably should be more back-of-mind for users. Instagram, for example, has a bar that goes across the _very_ top of the screen (which we could also augment with the words "Blocks synced ✔️"). Again, not necessary for this PR, but just thinking about a future improvement.



https://github.com/penumbra-zone/web/assets/1121544/1736c09e-adf4-4cdd-99fd-55f6b6838b30

File: packages/ui/components/ui/block-sync-status/syncing-state.tsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        status='in-progress'
+        value={(fullSyncHeight / latestKnownBlockHeight) * 100}
+        background={background}
+      />
+      <p className='font-mono text-sand'>
+        {fullSyncHeight}/{latestKnownBlockHeight}
+      </p>
+      <p
+        className={cn(
+          '-mt-1 font-mono text-sm text-sand transition-all duration-300',
```
Comment (jessepinho) [inline packages/ui/components/ui/block-sync-status/syncing-state.tsx:48]:
```suggestion
          'mt-1 font-mono text-sm text-sand transition-all duration-300',
```

**Review by jessepinho — APPROVED** [review]
LGTM. Looks like a couple things need to be renamed, especially the `LAST_BLOCK_SYNCED` IDB key and the `getfullSyncHeight` method (due to capitalization). Feel free to self-merge after those changes! Nice work.

File: apps/webapp/src/components/header/sync-status-section.tsx (hunk 5 above)
Comment (grod220) [reply apps/webapp/src/components/header/sync-status-section.tsx:6]: Ah, you are so right. Relocating to the top.

**Pull request merged**